Source/resizeablerectitem.cpp

```diff
@@ -109,7 +109,43 @@ void ResizeableRectItem::paint( QPainter* aPainter, const QStyleOptionGraphicsIt
     }
 
     // Draw the rect outline.
-    aPainter->drawRect( cRect );
+    aPainter->drawRect(cRect);
+
+    // Draw the object name on the rect if no image is provided
+    if (m_Pixmap.isNull())
+    {
+        const auto objectName = mMapObject->mObjectStructureType.c_str();
+        for (;;)
+        {
+            QFont f = aPainter->font();
+            QFontMetricsF fm(f);
+            QRectF textRect = fm.boundingRect(cRect, Qt::AlignCenter | Qt::TextWrapAnywhere, objectName);
+            auto size = f.pointSize();
+            if (size <= 0)
+            {
+                size = 8;
```

Comment:
if pointSize returns -1 then size becomes 8 and --size can never <=0 resulting in an inf loop

Reply:
have you read the new lines below? 
```c++
            if (textRect.width() > cRect.width() ||
                textRect.height() > cRect.height())
            {
                if (size > 1)
                {
                    f.setPointSize(--size);
                }
                else
                {
                    break;
                }
            }
            else
            {
                break;
            }```
is supposed to prevent that

Reply:
auto size = f.pointSize(); // returns -1 
if (size <= 0)
            {
                size = 8;
                }
 // size is now 8
if (size > 1) { f.setPointSize(--size); } 

// size is now 7

// loop again and repeat forever

Reply:
am i misunderstanding something? at the end of the for loop im doing ``aPainter->setFont(f);`` and at the next loop ``QFont f = aPainter->font();`` which means that ``auto size = f.pointSize();`` should return 7 now? 🤔

Reply:
https://doc.qt.io/qt-5/qfont.html states it can return -1 at any time. I think if it ever returns -1 then it will always return -1. 

`Returns the point size of the font. Returns -1 if the font size was specified in pixels.`

Reply:
alright i hope its fixed now :begging: